Ds3/Helpers/Ds3ClientHelpers.cs

```diff
@@ -199,6 +205,24 @@ public IEnumerable<Ds3Object> ListObjects(string bucketName, string keyPrefix)
             } while (isTruncated);
         }
 
+        public static readonly Func<Ds3Object, bool> FolderFilterPredicate = obj => !obj.Name.EndsWith("/");
+        public static readonly Func<Ds3Object, bool> ZeroLengthFilterPredicate = obj => obj.Size != 0;
+
+        public static IEnumerable<Ds3Object> FilterDs3Objects(IEnumerable<Ds3Object> objects,
```

Comment:
by using the were linq opertaion i am doing it lazy.
From the docs:
"This method is implemented by using deferred execution. The immediate return value is an object that stores all the information that is required to perform the action. The query represented by this method is not executed until the object is enumerated either by calling its GetEnumerator method directly or by using foreach in Visual C# or For Each in Visual Basic."